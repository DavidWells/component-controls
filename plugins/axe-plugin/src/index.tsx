/* eslint-disable react/display-name */
import React, { FC } from 'react';

import {
  StoryBlockContainer,
  StoryBlockContainerProps,
} from '@component-controls/blocks';
import { Spec } from 'axe-core';

import { BaseAllyBlock } from './components/BaseAllyBlock';

interface AxeAllyBlockOwmProps {
  axeOptions?: Spec;
}

export type AxeAllyBlockProps = AxeAllyBlockOwmProps & StoryBlockContainerProps;

/**
 * Story block container that displays displays the [axe](https://github.com/dequelabs/axe-core) ally test results
 */
export const AxeAllyBlock: FC<AxeAllyBlockProps> = ({
  axeOptions,
  ...props
}) => {
  return (
    <StoryBlockContainer {...props}>
      {({ story }) => (
        <BaseAllyBlock storyId={story?.id} options={axeOptions} />
      )}
    </StoryBlockContainer>
  );
};
