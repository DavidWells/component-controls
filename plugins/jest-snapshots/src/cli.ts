import path from 'path';
import fs from 'fs';
import { run as runJest } from 'jest-cli';
import {
  run as runCompiler,
  defaultCliArgs,
  cliArgs,
} from '@component-controls/webpack-compile/cli';
import { jestCliArgs } from './args';

const isCommandArg = (arg: string) =>
  arg.startsWith('--') || arg.startsWith('-');
export const run = async (): Promise<void> => {
  await runCompiler(jestCliArgs);
  const args = cliArgs(jestCliArgs);

  let testFolder =
    (args.parse().test as string) || path.resolve(process.cwd(), 'tests');
  if (!path.isAbsolute(testFolder)) {
    testFolder = path.resolve(process.cwd(), testFolder);
  }
  const allArgs = [...defaultCliArgs, ...jestCliArgs];
  const argv = process.argv.slice(2);
  const storiesFileName = 'stories.test.js';
  const jestArgs: string[] = [];
  const pushJestArg = (arg: string, len: number) => {
    const command = arg.substring(len);
    if (
      len &&
      allArgs.find(
        cliArg => command === (len === 2 ? cliArg.name : cliArg.options.alias),
      )
    ) {
      return true;
    } else {
      if (command.startsWith('jest-')) {
        jestArgs.push(`${'-'.repeat(len)}${command.substring(5)}`);
      } else {
        jestArgs.push(arg);
      }
      return false;
    }
  };
  for (let i = 0; i < argv.length; i += 1) {
    if (argv[i].startsWith('--')) {
      if (
        pushJestArg(argv[i], 2) &&
        i < argv.length - 1 &&
        !isCommandArg(argv[i + 1])
      ) {
        i += 1;
      }
    } else if (argv[i].startsWith('-')) {
      if (
        pushJestArg(argv[i], 1) &&
        i < argv.length - 1 &&
        !isCommandArg(argv[i + 1])
      ) {
        i += 1;
      }
    } else {
      pushJestArg(argv[i], 0);
    }
  }
  if (!fs.existsSync(testFolder)) {
    fs.mkdirSync(testFolder);
  }
  fs.copyFileSync(
    path.resolve(__dirname, '..', storiesFileName),
    path.resolve(testFolder, storiesFileName),
  );
  await runJest(jestArgs, testFolder);
  fs.unlinkSync(path.resolve(testFolder, storiesFileName));
};
