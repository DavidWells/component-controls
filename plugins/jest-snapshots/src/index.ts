import { loadStore } from '@component-controls/store';
import { getBundleName } from '@component-controls/webpack-compile';
import { Store } from '@component-controls/core';

export type RendererFnResult = Promise<any> | any;

export type RendererFn = (
  storyId: string,
  store: Store,
  options?: any,
) => Promise<RendererFnResult>;

export const runJestSnapshots = (renderer: RendererFn, bundleName?: string) => {
  const bundle = bundleName || getBundleName();
  const store = loadStore(require(bundle));
  Object.keys(store.docs).forEach(docId => {
    const doc = store.docs[docId];

    if (doc.stories && doc.stories.length) {
      const stories = doc.stories;
      describe(doc.title, () => {
        stories.forEach(storyId => {
          const story = store.stories[storyId];
          it(story.name, async () => {
            const tree = await renderer(storyId, store);
            expect(tree).toMatchSnapshot();
          });
        });
      });
    }
  });
};
