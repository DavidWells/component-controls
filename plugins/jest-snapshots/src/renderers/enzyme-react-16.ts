import { mount, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { Store } from '@component-controls/core';
import { RendererFn } from '../index';

configure({ adapter: new Adapter() });

export const render: RendererFn = (
  storyId: string,
  store: Store,
  options?: any,
) => {
  const renderFn = store.config.renderFn;
  if (renderFn) {
    const component = mount(renderFn(storyId, store, options));
    return toJson(component, { mode: 'deep' });
  }
  return undefined;
};
