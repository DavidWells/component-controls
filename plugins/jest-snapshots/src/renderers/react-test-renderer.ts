import renderer from 'react-test-renderer';
import { Store } from '@component-controls/core';
import { RendererFn } from '../index';

export const render: RendererFn = (
  storyId: string,
  store: Store,
  options?: any,
) => {
  const renderFn = store.config.renderFn;
  if (renderFn) {
    const component = renderer.create(renderFn(storyId, store, options));
    return component.toJSON();
  }
  return undefined;
};
