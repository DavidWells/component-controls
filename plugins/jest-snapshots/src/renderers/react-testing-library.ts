import { render as rtlRender, cleanup } from '@testing-library/react';

import { Store } from '@component-controls/core';
import { RendererFn } from '../index';

export const render: RendererFn = (
  storyId: string,
  store: Store,
  options?: any,
) => {
  const renderFn = store.config.renderFn;
  if (renderFn) {
    cleanup();
    let fragment: DocumentFragment | undefined = undefined;
    const { asFragment } = rtlRender(renderFn(storyId, store, options));
    fragment = asFragment();
    return fragment;
  }
  return undefined;
};
