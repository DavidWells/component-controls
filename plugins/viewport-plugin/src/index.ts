export * from './ViewportBlock';
export * from './ViewportPage';
