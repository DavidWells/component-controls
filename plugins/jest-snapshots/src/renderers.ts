import { render as reactTestRenderer } from './renderers/react-test-renderer';
import { render as enzymeReact16Renderer } from './renderers/enzyme-react-16';
import { render as rtlRenderer } from './renderers/react-testing-library';
import { RendererFn } from './index';

export const renderers: Record<string, RendererFn> = {
  react: rtlRenderer,
  rtr: reactTestRenderer,
  enzyme: enzymeReact16Renderer,
};
