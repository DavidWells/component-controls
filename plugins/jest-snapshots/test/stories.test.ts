import { renderers } from '../src/renderers';
import { runJestSnapshots } from '../dist/index';
runJestSnapshots(renderers.enzyme);
runJestSnapshots(renderers.react);
