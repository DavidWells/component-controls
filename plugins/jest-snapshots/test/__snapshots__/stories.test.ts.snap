// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blocks/Story Custom Title 1`] = `
Array [
  "",
  "",
  <div
    className="css-1xkyebe"
  >
    <div
      className="css-jxxswp"
      id=""
    >
      <div
        className="css-xhp9da"
      >
        <h3
          className="css-1p18zt3-BlockContainer"
        >
          <div
            className="css-tmrrvx"
          >
            My Story Title
            <a
              aria-label="Collapse this block"
              className="css-csemrf-BlockContainer"
              onClick={[Function]}
            >
              <div
                className="css-1ibxxpj"
              >
                <svg
                  aria-hidden="true"
                  className="octicon"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<path fill-rule=\\"evenodd\\" d=\\"M12.78 6.22a.75.75 0 010 1.06l-4.25 4.25a.75.75 0 01-1.06 0L3.22 7.28a.75.75 0 011.06-1.06L8 9.94l3.72-3.72a.75.75 0 011.06 0z\\"></path>",
                    }
                  }
                  fill="currentColor"
                  height={16}
                  role="img"
                  style={
                    Object {
                      "display": "inline-block",
                      "userSelect": "none",
                      "verticalAlign": "middle",
                    }
                  }
                  viewBox="0 0 16 16"
                  width={16}
                />
              </div>
            </a>
          </div>
        </h3>
      </div>
    </div>
    <div
      aria-hidden={false}
      className="rah-static rah-static--height-auto"
      style={
        Object {
          "height": "auto",
          "overflow": "visible",
        }
      }
    >
      <div
        style={Object {}}
      >
        <div
          className="css-154oin1"
          data-testid="story"
          id="blocks-core-story-plain--controls"
        >
          <div
            className="story-render-container css-dhmrnb"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Blocks/Story Custom Title 2`] = `
<DocumentFragment>
  <div
    class="css-1xkyebe"
  >
    <div
      class="css-jxxswp"
      id=""
    >
      <div
        class="css-xhp9da"
      >
        <h3
          class="css-1p18zt3-BlockContainer"
        >
          <div
            class="css-tmrrvx"
          >
            My Story Title
            <a
              aria-label="Collapse this block"
              class="css-csemrf-BlockContainer"
            >
              <div
                class="css-1ibxxpj"
              >
                <svg
                  aria-hidden="true"
                  class="octicon"
                  fill="currentColor"
                  height="16"
                  role="img"
                  style="display: inline-block; user-select: none; vertical-align: middle;"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <path
                    d="M12.78 6.22a.75.75 0 010 1.06l-4.25 4.25a.75.75 0 01-1.06 0L3.22 7.28a.75.75 0 011.06-1.06L8 9.94l3.72-3.72a.75.75 0 011.06 0z"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </a>
          </div>
        </h3>
      </div>
    </div>
    <div
      aria-hidden="false"
      class="rah-static rah-static--height-auto"
      style="height: auto; overflow: visible;"
    >
      <div>
        <div
          class="css-154oin1"
          data-testid="story"
          id="blocks-core-story-plain--controls"
        >
          <div
            class="story-render-container css-dhmrnb"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Blocks/Story Description 1`] = `
Array [
  "",
  "",
  <div
    className="css-1xkyebe"
  >
    <div
      className="css-jxxswp"
      id=""
    >
      <div
        className="css-xhp9da"
      >
        <h3
          className="css-1p18zt3-BlockContainer"
        />
      </div>
    </div>
    <div
      aria-hidden={false}
      className="rah-static rah-static--height-auto"
      style={
        Object {
          "height": "auto",
          "overflow": "visible",
        }
      }
    >
      <div
        style={Object {}}
      >
        <div
          className="css-154oin1"
          data-testid="story"
          id="id-of-story"
        >
          <div
            className="story-render-container css-dhmrnb"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Blocks/Story Description 2`] = `
<DocumentFragment>
  <div
    class="css-1xkyebe"
  >
    <div
      class="css-jxxswp"
      id=""
    >
      <div
        class="css-xhp9da"
      >
        <h3
          class="css-1p18zt3-BlockContainer"
        />
      </div>
    </div>
    <div
      aria-hidden="false"
      class="rah-static rah-static--height-auto"
      style="height: auto; overflow: visible;"
    >
      <div>
        <div
          class="css-154oin1"
          data-testid="story"
          id="id-of-story"
        >
          <div
            class="story-render-container css-dhmrnb"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Blocks/Story Iframe 1`] = `
Array [
  "",
  "",
  <div
    className="css-1xkyebe"
  >
    <div
      className="css-jxxswp"
      id=""
    >
      <div
        className="css-xhp9da"
      >
        <h3
          className="css-1p18zt3-BlockContainer"
        />
      </div>
    </div>
    <div
      aria-hidden={false}
      className="rah-static rah-static--height-auto"
      style={
        Object {
          "height": "auto",
          "overflow": "visible",
        }
      }
    >
      <div
        style={Object {}}
      >
        <div
          className="css-154oin1"
          data-testid="story"
          id="blocks-core-story-plain--controls"
        >
          <iframe
            style={
              Object {
                "border": 0,
                "height": undefined,
                "width": "100%",
              }
            }
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Blocks/Story Iframe 2`] = `
<DocumentFragment>
  <div
    class="css-1xkyebe"
  >
    <div
      class="css-jxxswp"
      id=""
    >
      <div
        class="css-xhp9da"
      >
        <h3
          class="css-1p18zt3-BlockContainer"
        />
      </div>
    </div>
    <div
      aria-hidden="false"
      class="rah-static rah-static--height-auto"
      style="height: auto; overflow: visible;"
    >
      <div>
        <div
          class="css-154oin1"
          data-testid="story"
          id="blocks-core-story-plain--controls"
        >
          <iframe
            style="border: 0px; width: 100%;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Blocks/Story Not Collapsible 1`] = `
Array [
  "",
  "",
  <div
    className="css-1xkyebe"
  >
    <div
      className="css-154oin1"
      data-testid="story"
      id="blocks-core-story-plain--controls"
    >
      <div
        className="story-render-container css-dhmrnb"
      />
    </div>
  </div>,
]
`;

exports[`Blocks/Story Not Collapsible 2`] = `
<DocumentFragment>
  <div
    class="css-1xkyebe"
  >
    <div
      class="css-154oin1"
      data-testid="story"
      id="blocks-core-story-plain--controls"
    >
      <div
        class="story-render-container css-dhmrnb"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Blocks/Story Overview 1`] = `
Array [
  "",
  "",
  <div
    className="css-1xkyebe"
  >
    <div
      className="css-jxxswp"
      id=""
    >
      <div
        className="css-xhp9da"
      >
        <h3
          className="css-1p18zt3-BlockContainer"
        />
      </div>
    </div>
    <div
      aria-hidden={false}
      className="rah-static rah-static--height-auto"
      style={
        Object {
          "height": "auto",
          "overflow": "visible",
        }
      }
    >
      <div
        style={Object {}}
      >
        <div
          className="css-154oin1"
          data-testid="story"
          id="blocks-core-story-plain--controls"
        >
          <div
            className="story-render-container css-dhmrnb"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Blocks/Story Overview 2`] = `
<DocumentFragment>
  <div
    class="css-1xkyebe"
  >
    <div
      class="css-jxxswp"
      id=""
    >
      <div
        class="css-xhp9da"
      >
        <h3
          class="css-1p18zt3-BlockContainer"
        />
      </div>
    </div>
    <div
      aria-hidden="false"
      class="rah-static rah-static--height-auto"
      style="height: auto; overflow: visible;"
    >
      <div>
        <div
          class="css-154oin1"
          data-testid="story"
          id="blocks-core-story-plain--controls"
        >
          <div
            class="story-render-container css-dhmrnb"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Blocks/Story Title 1`] = `
Array [
  "",
  "",
  <div
    className="css-1xkyebe"
  >
    <div
      className="css-jxxswp"
      id=""
    >
      <div
        className="css-xhp9da"
      >
        <h3
          className="css-1p18zt3-BlockContainer"
        />
      </div>
    </div>
    <div
      aria-hidden={false}
      className="rah-static rah-static--height-auto"
      style={
        Object {
          "height": "auto",
          "overflow": "visible",
        }
      }
    >
      <div
        style={Object {}}
      >
        <div
          className="css-154oin1"
          data-testid="story"
          id="blocks-core-story-plain--controls"
        >
          <div
            className="story-render-container css-dhmrnb"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Blocks/Story Title 2`] = `
<DocumentFragment>
  <div
    class="css-1xkyebe"
  >
    <div
      class="css-jxxswp"
      id=""
    >
      <div
        class="css-xhp9da"
      >
        <h3
          class="css-1p18zt3-BlockContainer"
        />
      </div>
    </div>
    <div
      aria-hidden="false"
      class="rah-static rah-static--height-auto"
      style="height: auto; overflow: visible;"
    >
      <div>
        <div
          class="css-154oin1"
          data-testid="story"
          id="blocks-core-story-plain--controls"
        >
          <div
            class="story-render-container css-dhmrnb"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
